GiniVision/Classes/Core/CapturedImagesStackView.swift

```diff
@@ -0,0 +1,178 @@
+        }
+    }()
+    var didTapImageStackButton: (() -> Void)?
+    fileprivate let stackCountCircleSize = CGSize(width: 25, height: 25)
+    fileprivate var imagesCount: Int = 0
+    
+    fileprivate lazy var thumbnailButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowRadius = 1
+        button.layer.shadowOpacity = 0.5
+        button.layer.shadowOffset = CGSize(width: -2, height: 2)
+        button.addTarget(self, action: #selector(thumbnailButtonAction), for: .touchUpInside)
+        
+        return button
+    }()
+    
+    lazy var thumbnailStackBackgroundView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .lightGray
+        view.isHidden = true
+        return view
+    }()
+    
+    fileprivate lazy var stackIndicatorLabel: UILabel = {
+        var label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = Colors.Gini.blue
+        return label
+    }()
+    
+    fileprivate lazy var stackIndicatorCircleView: UIView = {
+        var view = UIView(frame: .zero)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.frame.size = self.stackCountCircleSize
+        view.backgroundColor = .white
+        view.layer.cornerRadius = self.stackCountCircleSize.width / 2
+        view.layer.shadowRadius = 1
+        view.layer.shadowOpacity = 0.3
+        view.layer.shadowOffset = CGSize(width: -1, height: 1)
+        return view
+    }()
+    
+    fileprivate lazy var capturedImagesStackSubtitleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = NSLocalizedString("ginivision.camera.capturedImagesStackLabel",
+                                       bundle: Bundle(for: GiniVision.self),
+                                       comment: "label shown below images stack")
+        label.textAlignment = .center
+        label.textColor = .white
+        label.font = label.font.withSize(12)
+
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addSubview(thumbnailStackBackgroundView)
+        addSubview(thumbnailButton)
+        addSubview(stackIndicatorCircleView)
+        addSubview(capturedImagesStackSubtitleLabel)
+        stackIndicatorCircleView.addSubview(stackIndicatorLabel)
+        addConstraints()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("Use init(frame:) initializer instead")
+    }
+
+}
+
+// MARK: - Public methods
+
+extension CapturedImagesStackView {
+    func absoluteThumbnailFrame(from view: UIView) -> CGRect {
+        return convert(thumbnailButton.frame, to: view)
+    }
+    
+    func updateStackStatus(to status: State) {
```

Comment:
I'm not a fan of this interface. I have to create a State variable to change the stack view? It seems like an overkill and it's a bit confusing. 

I'd prefer something similar to the next method, `addImageToStack`.
The most common use case is to add or remove an image (maybe also reorder or replace).

Reply:
You are right, also does not make sense to have different ways of updating it.